doio.c

```diff
@@ -2214,36 +2214,37 @@ Perl_do_print(pTHX_ SV *sv, PerlIO *fp)
     else {
         STRLEN len;
         /* Do this first to trigger any overloading.  */
-        const char *tmps = SvPV_const(sv, len);
+        const U8 *tmps = (U8 *) SvPV_const(sv, len);
```

Comment:
For consistency the cast should be to `(const U8 *)`, since the rest of  SvPV_const() is a const pointer, you're casting const away just to add it back in. (makes me wish for C++ const_cast, static_cast etc)